hs_file_types/models/base.py

```diff
@@ -1189,33 +1280,47 @@ def remove_aggregation(self):
         # deleting the logical file object will not automatically delete the associated
         # metadata file object
         metadata = self.metadata if self.has_metadata else None
-        super(AbstractLogicalFile, self).delete()
+
+        # if we are removing a model program aggregation, then we need to set the
+        # metadata of all the associated model instances to dirty
+        if self.is_model_program:
```

Comment:
When I see things like this, I think AbstractLogicalFile is too aware of it's children.  Ideally, it should know nothing about the children.  Rather it should provide an interface that allows the logic to be separated into each child.